packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js

```diff
@@ -190,7 +190,11 @@ class LightCurveViewer extends Component {
   getAnnotationValues () {
     const props = this.props
     const annotations = (props.annotations && props.annotations.toJSON()) || {}
-    return (annotations[props.currentTask.taskKey] && [...annotations[props.currentTask.taskKey].value]) || []
+    if (this.isCurrentTaskValidForAnnotation()) {
+      return (annotations[props.currentTask.taskKey] && [...annotations[props.currentTask.taskKey].value]) || []
```

Comment:
Ah, this bit of sorcery shows where I struggled with MobX.

Basically, `getAnnotationValues()` needs to return an array `[{}]` of annotations for D3, but every time I tried to extract the data from the stores, I get a MobX map that D3 couldn't process.

```
console.log('+++ A: ', (annotations[props.currentTask.taskKey] && [...annotations[props.currentTask.taskKey].value]))
      console.log('+++ B: ', props.annotations.get(props.currentTask.taskKey) && props.annotations.get(props.currentTask.taskKey).value)
```
![screen shot 2019-01-22 at 14 09 51](https://user-images.githubusercontent.com/13952701/51541001-aae69a80-1e4f-11e9-9eb6-e91068385a86.png)
_Example of the data coming out as a Map instead of an array of Objects_

There's likely a better way than this brute force method of getting that array-of-annotations, (or at least there should be a way of streamlining the code to make it easier to read,) though so far I can't escape using `toJSON()`

Reply:
AH ok, yes. It's a major gotcha with MST that the data is wrapped in observables which are objects, so an observable array is an object... takes a bit to wrap your head around. So if you need a native array, how about?:
```
const { annotations } = this.props
const annotation = annotations.get(props.currentTask.taskKey)
if (this.isCurrentTaskValidForAnnotation()) return Array.from(annotation.value) || []
return []
```

The Classification store sets a default value for annotations, so the existential check shouldn't be necessary for that itself.

Reply:
👌 That looks good, I've used that with some minor modifications to check for the existence of `annotation`:
```
const { annotations, currentTask } = this.props
    const annotation = annotations.get(currentTask.taskKey)
    if (annotation && this.isCurrentTaskValidForAnnotation()) return Array.from(annotation.value) || []
    return []
```